com.sap.cloud.lm.sl.cf.core/src/test/java/com/sap/cloud/lm/sl/cf/core/parser/DockerInfoParserTest.java

```diff
@@ -0,0 +1,89 @@
+    public void setUp() {
+        dockerInfoParser = new DockerInfoParser();
+    }
+
+    @Test
+    public void testWithValidImageWithoutUserCredentials() {
+        List<Map<String, Object>> parameters = new ArrayList<>();
+        Map<String, Object> moduleParameters = new HashMap<>();
+        Map<String, String> dockerParameters = new LinkedHashMap<>();
+        String sampleImage = "cloudfoundry/test-app";
+        dockerParameters.put("image", sampleImage);
+        moduleParameters.put("docker", dockerParameters);
+        parameters.add(moduleParameters);
+
+        DockerInfo actualDockerInfo = dockerInfoParser.parse(parameters);
+        DockerInfo expectedDockerInfo = new DockerInfo(sampleImage);
+
+        assertEquals(expectedDockerInfo.getImage(), actualDockerInfo.getImage());
+        assertEquals(expectedDockerInfo.getDockerCredentials(), expectedDockerInfo.getDockerCredentials());
+    }
+
+    @Test
+    public void testWithValidImageAndCredentials() {
+        List<Map<String, Object>> parameters = new ArrayList<>();
+        Map<String, Object> moduleParameters = new HashMap<>();
+        Map<String, String> dockerParameters = new LinkedHashMap<>();
+        String sampleImage = "cloudfoundry/test-app";
+        String username = "someUsername";
+        String password = "somePassword";
+
+        dockerParameters.put("image", sampleImage);
+        dockerParameters.put("username", username);
+        dockerParameters.put("password", password);
+
+        moduleParameters.put("docker", dockerParameters);
+        parameters.add(moduleParameters);
+
+        DockerInfo actualDockerInfo = dockerInfoParser.parse(parameters);
+        DockerInfo expectedDockerInfo = new DockerInfo(sampleImage);
+        expectedDockerInfo.setDockerCredentials(new DockerCredentials(username, password));
+
+        assertEquals(expectedDockerInfo.getImage(), actualDockerInfo.getImage());
+
+        assertEquals(expectedDockerInfo.getDockerCredentials()
+            .getUsername(),
+            expectedDockerInfo.getDockerCredentials()
+                .getUsername());
+
+        assertEquals(expectedDockerInfo.getDockerCredentials()
+            .getPassword(),
+            expectedDockerInfo.getDockerCredentials()
+                .getPassword());
+    }
+
+    @Test
+    public void testWithoutArguments() {
+        List<Map<String, Object>> parameters = new ArrayList<>();
+        Map<String, Object> moduleParameters = new HashMap<>();
+        Map<String, String> dockerParameters = new LinkedHashMap<>();
+        moduleParameters.put("docker", dockerParameters);
+        parameters.add(moduleParameters);
+
+        DockerInfo actualDockerInfo = dockerInfoParser.parse(parameters);
+        DockerInfo expectedDockerInfo = new DockerInfo(null);
+
+        assertEquals(expectedDockerInfo.getImage(), actualDockerInfo.getImage());
+        assertEquals(expectedDockerInfo.getDockerCredentials(), expectedDockerInfo.getDockerCredentials());
```

Comment:
Here you're validating whether the expectedDockerInfo's credentials are equal to themselves.